Develop

Combo3zoom opened the pull request.

Combo3zoom committed:
add class SuperPowerGenerator

Combo3zoom committed:
Merge pull request #6 from Combo3zoom/feature/1-SuperPowerGenerator

add class SuperPowerGenerator

Combo3zoom committed:
add abstract class Warrior and other 4 subclass

Combo3zoom committed:
Merge pull request #7 from Combo3zoom/feature/2-Create_Warrior

add abstract class Warrior and other 4 subclass

Combo3zoom committed:
1) add Health/Armor/Damage super power
2) changed properties health/armor/damage from int to float
3) update self random system

Combo3zoom committed:
Merge pull request #8 from Combo3zoom/feature/3-super_power_warrion

realize super power and fix smt

Combo3zoom committed:
add properties HasAlive and StrikeCounter

Combo3zoom committed:
add class BattleArena

Combo3zoom committed:
Merge pull request #9 from Combo3zoom/feature/4-battle_arena

Feature/4 battle arena

Combo3zoom committed:
1) add menu(for battle)
2) fix default constructor in son class

Combo3zoom committed:
Merge pull request #10 from Combo3zoom/feature/5-menu

realization battle

Combo3zoom committed:
add method Count the digit

Combo3zoom committed:
Merge pull request #12 from Combo3zoom/featureLINQ/1-CountTheDigit

add method Count the digit

Combo3zoom committed:
add method ReduceButGrow

Combo3zoom committed:
Merge pull request #13 from Combo3zoom/featureLINQ/2-ReduceButGrow

add method ReduceButGrow

Combo3zoom committed:
add method Square(n) sum
squares and sums all elements

Combo3zoom committed:
Merge pull request #14 from Combo3zoom/featureLINQ/3-Square(n)Sum

add method Square(n) sum

Combo3zoom committed:
add method ReverseAndRotate

Combo3zoom committed:
snapped code

Combo3zoom committed:
Merge pull request #15 from Combo3zoom/featureLINQ/4-ReverseAndRotate

add method ReverseAndRotate

Combo3zoom committed:
add method SumofOddNumbers

Combo3zoom committed:
Merge branch 'develop' into featureLINQ/5-SumOfOddNumbers

# Conflicts:
#	ExoftCamp.LINQ/Program.cs
#	ExoftCamp.LINQ/TaskLINQ.cs

Combo3zoom committed:
Merge pull request #16 from Combo3zoom/featureLINQ/5-SumOfOddNumbers

snapped code

Combo3zoom committed:
fix LINQ library

Combo3zoom committed:
optimise and fix bugs

Combo3zoom committed:
Merge pull request #17 from Combo3zoom/bugFix

optimise and fix bugs

Combo3zoom committed:
four customs operators have been added

Combo3zoom committed:
Merge pull request #18 from Combo3zoom/feature/6-_operators

four customs operators have been added

Combo3zoom committed:
upgrade console(unlimit battle)

Combo3zoom committed:
Merge pull request #19 from Combo3zoom/feature/7-upgrade_console

upgrade console(unlimit battle)

YevhenVolynets commented on ExoftCamp.Game/Program.cs:
```diff
@@ -1,3 +1,51 @@
+                    case "2":
+                        while (warriors.Count != indexBattle)
+                        {
+                            var lastTwoWarriors = 
```
The exception is thrown if I create only one warrior.

YevhenVolynets commented on ExoftCamp.Game/Program.cs:
```diff
@@ -1,3 +1,51 @@
+                        {
+                            var lastTwoWarriors = 
+                                BattleArena.Duel(warriors[indexBattle++], warriors[indexBattle++]);
+                            Console.WriteLine($"Winner {lastTwoWarriors[0]}" +
```
A situation where there will be no winner is possible.

YevhenVolynets commented on WarriorLibrary/BattleArena.cs:
```diff
@@ -0,0 +1,31 @@
+            : winner == secondBaseWarrior
+                ? secondBaseWarrior
+                : null;
+        return new List<BaseWarrior>{winner!, looser!};
```
It is not the best solution.  1. You can have a situation when you'll have no winner and no looser.
2. Why do you need to return the list which contains only 2 elements every time? You have the same data by references in the class which calls this method. The order in the list is not clear that the first one in the list is the winner.

YevhenVolynets commented on WarriorLibrary/Models/BaseWarrior.cs:
```diff
@@ -0,0 +1,121 @@
+        }
+    }
+
+    public bool HasAlive => Health > 0;
```
*IsAlive

YevhenVolynets commented on ExoftCamp.Game/HelperBattle.cs:
```diff
@@ -0,0 +1,66 @@
+{
+    public static BaseWarrior CreateWarrior(string message)
+        {
+            BaseWarrior? warrior = null;
```
Why is it nullable?

YevhenVolynets commented on ExoftCamp.Game/HelperBattle.cs:
```diff
@@ -0,0 +1,66 @@
+                    case 2:
+                        Console.WriteLine("<------------------------------------>");
+                        Console.WriteLine("Enter type warrior:\n 1 - swordmen,\n 2 - spearmen,\n 3 - bowmen,\n 4 - crossbowmen");
+                        var userChooseTypeWarrior = uint.Parse(Console.ReadLine()!);
```
What happens if I select 5, "blalbla", null?

YevhenVolynets commented on ExoftCamp.Game/HelperBattle.cs:
```diff
@@ -0,0 +1,66 @@
+                        switch (userChooseTypeSuperPower)
+                        { 
+                            case 1:
+                                warrior!.AddSuperPower(new HealthSuperPower(), countPower);
```
What happens if I try to add power without previously created warrior?

Combo3zoom committed:
fix bags
1) upgrade console
2) add validation
3) an unlimited number of fighters can fight in the arena

Combo3zoom committed:
Merge pull request #20 from Combo3zoom/Bugfix

fix bags
